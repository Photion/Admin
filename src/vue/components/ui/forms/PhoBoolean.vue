<template>
  <div class="pho-switch-wrapper">
    <div class="pho-switch-container">
      <div class="pho-switch">
        <input
          type="checkbox"
          :id="id.unique"
          :value="modelValue"
          @input="onInput"
          :name="name"
          :cy="id.family" />
        <label :for="id.unique" class="pho-switch-circle"></label>
      </div>
      <label :for="id.unique" class="pho-switch-text">{{ label }}</label>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

import { useId, componentProps } from '~/src/vue/components/shared';
import { getFormProps } from '~/src/vue/components/ui/forms/forms';

export default defineComponent({

  props: {
    ...componentProps,
    ...getFormProps(Boolean),
  },

  setup(props, context) {
    const onInput = ($event: InputEvent) => {
      const target = $event.target as HTMLInputElement;

      return context.emit('update:modelValue', target.checked);
    };

    return {
      id: useId('field', props),
      onInput,
    };
  },

});
</script>
