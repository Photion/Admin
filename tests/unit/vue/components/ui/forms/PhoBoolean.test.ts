import { mount } from '@vue/test-utils';

import PhoBoolean from '~/src/vue/components/ui/forms/PhoBoolean.vue';

const getComponent = <T>(props: T) => {
  return mount(PhoBoolean, {
    props: {
      name: 'pho-boolean-comp',
      uuid: '65441ec3-a9fa-4e8c-b1e5-79d7d001e634',
      modelValue: false,
      ...props,
    },
  });
};

describe('unit.vue.components.ui.forms.PhoBoolean', () => {

  it('Mounts', () => {
    const wrapper = getComponent({});
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Sets the attributes', () => {
    const wrapper = getComponent({});
    const attributes = wrapper.get('[cy]').attributes();

    expect(attributes).toMatchSnapshot();
  });

  it('Renders the label', async () => {
    const wrapper = getComponent({ label: 'Magic Switch' });
    expect(wrapper.element).toMatchSnapshot();
  });

});
