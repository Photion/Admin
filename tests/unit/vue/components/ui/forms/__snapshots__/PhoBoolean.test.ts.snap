// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit.vue.components.ui.forms.PhoBoolean Mounts 1`] = `
<div
  class="pho-switch-wrapper"
>
  <!--v-if-->
  <div
    class="pho-switch-container"
  >
    <div
      class="pho-switch"
    >
      <input
        cy="field:pho-boolean-comp"
        id="field:pho-boolean-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
        name="pho-boolean-comp"
        type="checkbox"
        value="false"
      />
      <label
        class="pho-switch-circle"
        for="field:pho-boolean-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
      />
    </div>
    <!--v-if-->
  </div>
</div>
`;

exports[`unit.vue.components.ui.forms.PhoBoolean Renders the label (inline) 1`] = `
<div
  class="pho-switch-wrapper"
>
  <!--v-if-->
  <div
    class="pho-switch-container"
  >
    <div
      class="pho-switch"
    >
      <input
        cy="field:pho-boolean-comp"
        id="field:pho-boolean-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
        name="pho-boolean-comp"
        type="checkbox"
        value="false"
      />
      <label
        class="pho-switch-circle"
        for="field:pho-boolean-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
      />
    </div>
    <label
      class="pho-switch-text"
      for="field:pho-boolean-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    >
      Magic Switch
    </label>
  </div>
</div>
`;

exports[`unit.vue.components.ui.forms.PhoBoolean Renders the label (stacked) 1`] = `
<div
  class="pho-switch-wrapper"
>
  <div>
    <label
      class="pho-label"
      for="field:pho-boolean-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    >
      Magic Switch
    </label>
  </div>
  <div
    class="pho-switch-container"
  >
    <div
      class="pho-switch"
    >
      <input
        cy="field:pho-boolean-comp"
        id="field:pho-boolean-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
        name="pho-boolean-comp"
        type="checkbox"
        value="false"
      />
      <label
        class="pho-switch-circle"
        for="field:pho-boolean-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
      />
    </div>
    <!--v-if-->
  </div>
</div>
`;

exports[`unit.vue.components.ui.forms.PhoBoolean Sets the attributes 1`] = `
Object {
  "cy": "field:pho-boolean-comp",
  "id": "field:pho-boolean-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634",
  "name": "pho-boolean-comp",
  "type": "checkbox",
  "value": "false",
}
`;
